[14.0][MIG] sentry: migrate sentry-raven to new api sentry-sdk

paradoxxxzero opened the pull request:
This is a migration done by cherry-picking https://github.com/OCA/server-tools/pull/1942 on top of current 14.0.
This superseeds #2223

fernandahf committed:
[IMP] sentry: migrate sentry-raven to new api sentry-sdk

legalsylvain commented on sentry/__manifest__.py:
```diff
@@ -3,16 +3,25 @@
     "name": "Sentry",
     "summary": "Report Odoo errors to Sentry",
-    "version": "14.0.1.0.2",
+    "version": "14.0.1.0.0",
```
that version is incorrect.

paradoxxxzero replied:
Good catch

legalsylvain commented on sentry/__manifest__.py:
```diff
@@ -3,16 +3,25 @@
-    "Odoo Community Association (OCA)",
+    "Odoo Community Association (OCA),"
+    "Vauxoo",
+    "maintainers": ["barsi", "naglis", "versada", "moylop260", "fernandahf"],
```
I think there are no rules regarding maintainers quantity, but maybe 5 maintainers is a little too much ? 

CC : @OCA/community-maintainers

legalsylvain reviewed (commented):
partial review. 

thanks for porting this must-have update.

paradoxxxzero committed:
[FIX] sentry: Coerce numerical config values

paradoxxxzero committed:
[FIX] sentry: Patch odoo.service.server wsgi app

Because post_load is called after odoo.service.server start
It has already registered the unpatched   odoo.service.wsgi_server.application
So patch it here too.
This enables wsgi performance reporting with sentry_traces_sample_rate

paradoxxxzero committed:
[FIX] sentry: warn about sentry_transport only if it is specified in config

paradoxxxzero commented:
Thanks for the partial review.

I added some more fixes as explained in commits, most notably I added one more wsgi patch because as it is the `SentryWsgiMiddleware` was simply ignored by odoo. This hopefully will fix the performance monitoring.
Of course I'll rebase after review if it is deemed necessary.

andreschenkels commented:
I reviewed the module and it seems to be working fine. Nice information and no longer depending on the raven module!
Please merge

hparfr reviewed (approved):
I'm using it in production since weeks

LGTM

hparfr commented:
Can you please rebase ?

andreschenkels commented:
@legalsylvain  @paradoxxxzero can we move this forward? This should also be ported to v13 and v15.

can I help in any way?

dnplkndll commented:
using the traces_sample_rate can context be added to the transaction to reproduce the slow responses?

for instance a use case is if you have known long running calls, like to a shipping api, would like to filter the results to call not made with that certain action. Error traces have all the data. the performance samples do not. 

POST /web/dataset/call_button

<img width="358" alt="image" src="https://user-images.githubusercontent.com/6782997/161801568-f3310c45-5e76-4779-9921-148e32f5143b.png">

would this be a good place to have an option like inherit_id="web.assets_backend"
```
                <script
                    src="https://browser.sentry-cdn.com/6.16.1/bundle.min.js"
                    integrity="sha384-WkFzsrcXKeJ3KlWNXojDiim8rplIj1RPsCbuv7dsLECoXY8C6Cx158CMgl+O+QKW"
                    crossorigin="anonymous"
                ></script>
                <script
                    src="https://browser.sentry-cdn.com/6.16.1/bundle.tracing.min.js"
                    integrity="sha384-hySah00SvKME+98UjlzyfP852AXjPPTh2vgJu26gFcwTlZ02/zm82SINaKTKwIX2"
                    crossorigin="anonymous"
                ></script>
                <script>
                Sentry.init({
                    dsn: "<t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.sentry_dsn', '14.0')",
                    release: "<t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.version', '14.0')"/>",
                    integrations: [new Sentry.Integrations.BrowserTracing()],
                    tracesSampleRate: <t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.sentry_sample_Rate', '0.2')"/>,
                    sendClientReports: false,
                });
                </script>
```

thapakazi commented:
+1, what needs to be done on this to get it merged ?

dreispt commented:
We need a green TravisCI build. That's blocking this right now.

paradoxxxzero committed:
[FIX] sentry: Fix bad ported version

hparfr commented:
Can we merge ?

dnplkndll commented:
@paradoxxxzero do you have any idea how to get the header set? there are a number of integration examples now. But not clear to me the best approach here. 

https://develop.sentry.dev/sdk/performance/#header-sentry-trace

hparfr commented:
@schiggi @dnplkndll @thapakazi 

Can you please mark your review status on this PR ?

sebastienbeau commented:
/ocabot merge patch
